### ISSUE TITLE
test_fft_input_modification_cpu should be skipped when no MKL

### ISSUE THREAD

**Issue by zasdfgbnm** [pytorch/pytorch#37156]
## 🐛 Bug

On my machine, the following test always fails:

```
test/test_torch.py::TestTorchDeviceTypeCPU::test_fft_input_modification_cpu
```

with error message:

```
=================================================================================================================================== FAILURES ===================================================================================================================================
____________________________________________________________________________________________________________ TestTorchDeviceTypeCPU.test_fft_input_modification_cpu ____________________________________________________________________________________________________________

self = <test_torch.TestTorchDeviceTypeCPU testMethod=test_fft_input_modification_cpu>, device = 'cpu'

    @skipIfRocm
    def test_fft_input_modification(self, device):
        # FFT functions should not modify their input (gh-34551)
    
        signal = torch.ones((2, 2, 2), device=device)
        signal_copy = signal.clone()
>       spectrum = torch.fft(signal, 2)
E       RuntimeError: fft: ATen not compiled with MKL support

test/test_torch.py:12222: RuntimeError

```

## To Reproduce

Build PyTorch without MKL and run the test

cc @ezyang @gchanan @zou3519

**Comment by ezyang** [comment]
@zhangguanheng66 what's the reasoning for categorizing this as high priority?

**Comment by zhangguanheng66** [comment]
@ezyang I think this is related to our CI tests and it's easy to fix.

**Comment by peterbell10** [comment]
Already fixed by #36680
